library/public/engine.h

```diff
@@ -2,7 +2,18 @@
  * @class   engine
  * @brief   Class used to create instance of other classes
  *
- * TODO improve doc
+ * The main class of the libf3d to create all other instances
+ * Configured on creation using binary flags, then all objects
+ * can be accessed through their getter.
+ *
+ * Example usage:
+ *
+ *  f3d::engine eng(f3d::engine::CREATE_WINDOW | f3d::engine::CREATE_INTERACTOR);
```

Comment:
I think there are special tags for code in doxygen

Reply:
I wil add them buyt am still unsure of the way we will handle API documentation. I hate doxygen with a passion.